token-client/src/main/java/com/sap/cloud/security/xsuaa/client/SpringServiceManagerService.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * SPDX-FileCopyrightText: 2018-2021 SAP SE or an SAP affiliate company and Cloud Security Client Java contributors
+ * <p>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.sap.cloud.security.xsuaa.client;
+
+import com.sap.cloud.security.config.OAuth2ServiceConfiguration;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.*;
+import org.springframework.web.client.RestOperations;
+import org.springframework.web.client.RestTemplate;
+
+import javax.annotation.Nullable;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+
+public class SpringServiceManagerService implements ServiceManagerService {
```

Comment:
I haven't found XsuaaOAuth2SMServiceTest for it, only usage in spring-xsuaa. Maybe we can force spring-xsuaa to use HttpClient from apache as they require it for X.509 as well, then we need only a good concept on how they can customize the rest client with their own. Optimally only once for all 👍